src/transaction/mod.rs

```diff
@@ -14,16 +14,45 @@ pub use self::requests::Scanner;
 pub use self::transaction::{Snapshot, Transaction, TxnInfo};
 pub use super::rpc::Timestamp;
 
+use kvproto::kvrpcpb;
+
 use crate::{Key, Value};
 
 mod client;
 mod requests;
 #[allow(clippy::module_inception)]
 mod transaction;
 
+#[derive(Debug, Clone)]
 pub enum Mutation {
-    Put(Key, Value),
-    Del(Key),
-    Lock(Key),
-    Rollback(Key),
+    Put(Value),
+    Del,
+    Lock,
+}
+
+impl Mutation {
+    fn with_key(self, key: impl Into<Key>) -> kvrpcpb::Mutation {
```

Comment:
I think this will only ever be called with `Key`s so you can skip the `impl Into`.